domains/client/domain-operator/src/domain_block_processor.rs

```diff
@@ -977,6 +979,58 @@ where
     }
 }
 
+/// Generate MMR proof for the block `to_prove` in the current best fork. The returned proof
+/// can be later used to verify stateless (without query offchain MMR leaf) and extract the state
+/// root at `to_prove`.
+// TODO: remove `dead_code` after it is used in fraud proof generation
+#[allow(dead_code)]
+pub(crate) fn generate_mmr_proof<CClient, CBlock>(
```

Comment:
will we be using this in the relayer while generating MMR prrofs for XDM ?

Reply:
We can use this for generating MMR proofs for the consensus XDM, but there are some slight differences between the proof of the consensus chain and the domain chain, so we still need to keep 2 versions of this anyway, I just separate them as XDM vs FP instead of consensus vs domain.